shell/src/main/java/alluxio/cli/ValidateEnv.java

```diff
@@ -0,0 +1,392 @@
+        builder.redirectInput(ProcessBuilder.Redirect.INHERIT);
+        Process process = builder.start();
+        process.waitFor();
+        return process.exitValue() == 0;
+      } catch (IOException e) {
+        System.err.format("Unable to validate on node %s: %s.%n", node, e.getMessage());
+        return false;
+      }
+
+    } else {
+      System.err.format("Unable to reach ssh port 22 on node %s.%n", node);
+      return false;
+    }
+  }
+
+  // run validation tasks in local environment
+  private static boolean validateLocal(String target, String name) {
+    int validationCount = 0;
+    int failureCount = 0;
+    Collection<ValidationTask> tasks = TARGET_TASKS.get(target);
+    for (ValidationTask task : tasks) {
+      String taskName = TASK_NAMES.get(task);
+      if (name != null && !taskName.startsWith(name)) {
+        continue;
+      }
+
+      System.out.format("Validating %s...", taskName);
+      if (task.validate()) {
+        System.out.println("OK");
+      } else {
+        System.out.println("Failed");
+        failureCount++;
+      }
+
+      validationCount++;
+    }
+
+    if (failureCount > 0) {
+      System.out.format("Validation failed. Total failures: %d.%n", failureCount);
+      return false;
+    }
+
+    if (validationCount == 0) {
+      System.out.format("No validation task matched name \"%s\".%n", name);
+      return false;
+    }
+
+    System.out.println("Validation succeeded.");
+    return false;
+  }
+
+  private static boolean validateWorkers(String name) throws InterruptedException {
+    return validateRemote(readNodeList("workers"), "worker", name);
+  }
+
+  private static boolean validateMasters(String name) throws InterruptedException {
+    return validateRemote(readNodeList("masters"), "master", name);
+  }
+
+  /**
+   * Prints the help message.
+   *
+   * @param message message before standard usage information
+   */
+  public static void printHelp(String message) {
+    System.err.println(message);
+    HelpFormatter help = new HelpFormatter();
+    help.printHelp(USAGE, OPTIONS);
+  }
+
+  /**
+   * Validate environment.
+   *
+   * @param args list of arguments
+   * @return 0 on success, 1 on failures
+   */
+  public static int validate(String... args) throws InterruptedException {
+    CommandLineParser parser = new DefaultParser();
+    CommandLine cmd;
+
+    try {
+      cmd = parser.parse(OPTIONS, args, true /* stopAtNonOption */);
+    } catch (ParseException e) {
+      printHelp("Unable to parse input args: " + e.getMessage());
+      return 1;
+    }
+
+    Preconditions.checkNotNull(cmd, "Unable to parse input args.");
+    args = cmd.getArgs();
+    if (args.length < 1) {
+      printHelp("Target not specified.");
+      return 1;
+    }
+
+    if (args.length > 2) {
+      printHelp("More arguments than expected.");
+      return 1;
+    }
+
+    String target = args[0];
+    String name = args.length > 1 ? args[1] : null;
```

Comment:
The command will iterate through all tasks and check whether the name matches any of them. It will print error message if none of the tasks match the name.